sdk/identity/identity/src/credentials/managedIdentityCredential/fabricMsi.ts

```diff
@@ -100,16 +113,19 @@ export const fabricMsi: MSI = {
       ].join(" ")
     );
 
-    return msiGenericGetToken(
-      identityClient,
-      prepareRequestOptions(scopes, clientId),
-      expiresInParser,
-      getTokenOptions,
-      new https.Agent({
-        // This is necessary because Service Fabric provides a self-signed certificate.
-        // The alternative path is to verify the certificate using the IDENTITY_SERVER_THUMBPRINT env variable.
-        rejectUnauthorized: false
-      })
-    );
+    const request = createPipelineRequest({
+      abortSignal: getTokenOptions.abortSignal,
+      ...prepareRequestOptions(scopes, clientId),
+      allowInsecureConnection: true
+    });
+
+    request.agent = new https.Agent({
+      // This is necessary because Service Fabric provides a self-signed certificate.
+      // The alternative path is to verify the certificate using the IDENTITY_SERVER_THUMBPRINT env variable.
+      rejectUnauthorized: false
+    });
```

Comment:
This makes me very nervous. Can we try to validate the service fabric cert instead? Is there some kind of technical reason that the cert is self-signed instead of using a real trusted cert?

Reply:
Service fabric MSI certificates are self-signed since they’re only meant to be reached form the internal network. We have in our roadmaps to strengthen security of this authentication flow, however across languages we have agreed to skip the verification of the certificates. This is a local-network (cluster) endpoint, so it’s not as unsafe as a public domain endpoint, if that helps.

Reply:
If that's what folks have agreed to do, then I suppose that's OK, but I really wish we could at least check the thumbprint against a list or something. If not, then I guess it's just a limitation of the environment, but still makes me nervous.